### PR TITLE
Sdma fixes (take1)

### PR DESCRIPTION
**Pull request by dbaluta**
SDMA needs some refactoring to make the code cleaner and easier to undertand.

This is the first pull request with this goal:
- fix SDMA scripts
- better isolate channel 0.

### CODE REVIEW

**Comment by dbaluta** [comment]
@paulstelian97 I will be setting few PRs this week for refactoring SDMA.

File: src/drivers/imx/sdma.c [hunk 1]
```diff
@@ -480,31 +480,16 @@ static void sdma_channel_put(struct dma_chan_data *channel)
...
-	/* If channel is event driven, allow it to run by setting HOSTOVR.
-	 * If it's manually controlled, kickstart it by writing to SDMA_HSTART.
-	 */
-	if (pdata->hw_event != -1)
-		dma_reg_update_bits(channel->dma, SDMA_HOSTOVR,
-				    BIT(channel->index), BIT(channel->index));
-	else
-		dma_reg_write(channel->dma, SDMA_HSTART, BIT(channel->index));
+	dma_reg_update_bits(channel->dma, SDMA_HOSTOVR,
+			    BIT(channel->index), BIT(channel->index));
```
Comment (paulstelian97) [inline src/drivers/imx/sdma.c:492]: If we will want to support memory-to-memory transfers or use this as a host DMA we will need to revert this portion in the future. Perhaps a comment would be warranted to remind us of this.

Comment (dbaluta) [reply src/drivers/imx/sdma.c:492]: @paulstelian97 I think you have a good point. This has nothing to do with channel 0. Will update.

Comment (dbaluta) [reply src/drivers/imx/sdma.c]: @paulstelian97 done, kept that part. Thanks for review.

File: src/drivers/imx/sdma.c [hunk 2]
```diff
@@ -486,16 +486,10 @@ static int sdma_start(struct dma_chan_data *channel)
...
-		/* rest of the channels are prepared first */
-		if (channel->status != COMP_STATE_PREPARE &&
-		    channel->status != COMP_STATE_SUSPEND)
-			return -EINVAL;
-	} else {
-		/* channel 0 doesn't get set_config set on */
-		if (channel->status != COMP_STATE_READY)
-			return -EINVAL;
-	}
+	/* rest of the channels are prepared first */
```
Comment (paulstelian97) [inline src/drivers/imx/sdma.c:489]: This one comment may be unnecessary as it will lack context after the change.

File: src/drivers/imx/sdma.c [hunk 3]
```diff
@@ -208,6 +208,8 @@ static void sdma_init_c0(struct dma *dma)
 	pdata->ccb = sdma_pdata->ccb_array;
 	pdata->hw_event = -1;
 	dma_chan_set_data(c0, pdata);
+
+	dma_reg_write(dma, SDMA_CHNPRI(0), SDMA_MAXPRI);
```
Comment (paulstelian97) [inline src/drivers/imx/sdma.c:212]: I think commit title should specify that it's SDMA on this one as well?

Comment (dbaluta) [reply src/drivers/imx/sdma.c:212]: sorry, i don't get what you mean.

File: src/drivers/imx/sdma.c (hunk 2 above)
Comment (dbaluta) [reply src/drivers/imx/sdma.c:489]: yes, will remove.

File: src/drivers/imx/sdma.c (hunk 3 above)
Comment (paulstelian97) [reply src/drivers/imx/sdma.c:212]: You just mentioned drivers: imx: without the "sdma:" prefix on this last commit.

Comment (dbaluta) [reply src/drivers/imx/sdma.c:212]: I see. Thanks. Will fix.

**Comment by dbaluta** [comment]
SOFCI TEST

**Comment by dbaluta** [comment]
@jajanusz @mmaka1 care to have a look? It looks @lgirdwood is off :)

**Pull request merged**